src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -1275,8 +1282,18 @@ def _app_deploy(client, resource_group, service, app, name, version, path, runti
                 main_entry=None,
                 target_module=None,
                 no_wait=False,
-                file_type="Jar",
+                file_type=None,
                 update=False):
+    if file_type is None:
+        # update means command is az spring-cloud app deploy xxx
+        if update:
+            raise RequiredArgumentMissingError("One of the following arguments are required: '--artifact-path' or '--source-path'")
+        # if command is az spring-cloud app deployment create xxx, create default deployment
+        else:
+            logger.warning("Creating default deployment without artifact/source folder. Please specify the --artifact-path/--source-path argument explicitly if needed.")
+            default_deployment_resource = _default_deployment_resource_builder(cpu, memory, env, jvm_options, runtime_version, instance_count)
+            return sdk_no_wait(no_wait, client.deployments.begin_create_or_update,
+                               resource_group, service, app, name, default_deployment_resource)
```

Comment:
What will happen when both `file_type` and `update` are `None`?

Reply:
If both file_type and update are None, the code will go to else logic in line 1292. It won't happen because the method is called with update value (only true or false) until now.

Reply:
OK